python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -478,7 +478,7 @@ def write_parquet(
         )
 
     dict_policy = (
-        cudf_io_types.dictionary_policy.ALWAYS
+        cudf_io_types.dictionary_policy.ADAPTIVE
         if use_dictionary
```

Comment:
Thoughts on expanding this parameter to allow selecting ALWAYS?
It could be a string or a bool, where the bool keeps the behavior implemented here (true->adaptive, false->never). A string would map 1:1 to libcudf options.
CC @galipremsagar

Reply:
I agree with this but perhaps with an enum instead of strings?

Reply:
Would you want me to fold this into #15613? It's going to hit the same parts of the python code.

Reply:
I'd like to keep the policy change isolated in a separate PR, because it's a bit on the risky side.

Reply:
I'm not sure about enums. Looking at the current Python APIs, e.g.
```
def write_parquet(
    table,
    object filepaths_or_buffers,
    object index=None,
    object compression="snappy",
    object statistics="ROWGROUP",
    object metadata_file_path=None,
    object int96_timestamps=False,
    object row_group_size_bytes=_ROW_GROUP_SIZE_BYTES_DEFAULT,
    object row_group_size_rows=None,
    object max_page_size_bytes=None,
    object max_page_size_rows=None,
    object partitions_info=None,
    object force_nullable_schema=False,
    header_version="1.0",
    use_dictionary=True,
):
```
There's a lot of use of strings for parameters that could be enums. Would like to hear from a Python expert.

Reply:
Strings are fine as well but +1 on would like to hear from a Python expert on which would be a better choice.

Reply:
FWIW pyarrow uses strings for similar parameters.

Reply:
I think it's okay to leave as is. This is the docstring from pyarrow:


<img width="837" alt="Screenshot 2024-05-07 at 2 41 21 PM" src="https://github.com/rapidsai/cudf/assets/11664259/ed516621-f10c-4e80-b09e-761b08950eb0">


This means we just have to update our docstring of `use_dictionary` to say we will try to encode as dict until a certain size, and beyond that we leave it as plain...something like that.